nengo_gui/static/datastore.js

```diff
@@ -146,3 +139,177 @@ Nengo.DataStore.prototype.get_last_data = function() {
     }
     return shown;
 }
+
+
+Nengo.VariableDataStore = function(dims, sim, synapse){
```

Comment:
Actually `EnlargeableDataStore` might be better (as it doesn't allow shrinking).
(I also used `GrowableArray` for [something similar with Numpy arrays before](https://github.com/jgosmann/goppy/blob/master/goppy/growable.py).